### PR TITLE
toggle RTT mode after RAM initialization

### PR DESCRIPTION
**Pull request by japaric**
depends on knurling-rs/defmt#174
closes knurling-rs/defmt#133

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -333,10 +333,19 @@ fn notmain() -> Result<i32, anyhow::Error> {
 
         log::debug!("starting device");
         if core.get_available_breakpoint_units()? == 0 {
-            log::warn!("device doesn't support HW breakpoints; HardFault will NOT make `probe-run` exit with an error code");
-        } else {
-            core.set_hw_breakpoint(vector_table.hard_fault & !THUMB_BIT)?;
+            bail!("device without HW breakpoints are not supported");
```
Comment (japaric) [inline src/main.rs:336]: FIXME: we should only bail if `rtt_addr` is `Some`

File: src/main.rs [hunk 2]
```diff
@@ -933,6 +943,7 @@ fn rtt_and_heap_info_from(
         };
 
         match name {
+            "main" => main = Some(symbol.address() as u32 & !THUMB_BIT),
```
Comment (japaric) [inline src/main.rs:946]: I don't think this symbol is guaranteed to exist in the final binary so we may want to do something else

Comment (jonas-schievink) [reply src/main.rs:946]: At least when using cortex-m-rt it will always exist, I think

**Review by jonas-schievink — COMMENTED** [review]
Implementation looks good, would be good to resolve the FIXME before merging though

**Pull request merged**